packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -740,6 +756,95 @@ export function runConnectedCallback(vm: VM) {
     }
 }
 
+function setupContext(vm: VM) {
+    const contextKeys = getContextKeys();
+
+    if (isUndefined(contextKeys)) {
+        return;
+    }
+
+    const { connectContext } = contextKeys;
+    const { component, renderer } = vm;
+    const enumerableKeys = keys(getPrototypeOf(component));
+    const contextfulFieldsOrProps = ArrayFilter.call(
+        enumerableKeys,
+        (propName) => (component as any)[propName]?.[connectContext]
+    );
```

Comment:
I think we discussed doing it in a separate PR (which is totally fine) but we'll want to change this to use a "trusted context participate" scheme, similar to "trusted signals."